pkg/addressbook/addressbook.go

```diff
@@ -17,7 +17,12 @@ import (
 
 const keyPrefix = "addressbook_entry_"
 
-var _ GetPutter = (*store)(nil)
+var _ Interface = (*store)(nil)
+
+type Interface interface {
+	GetPutter
+	Overlays() ([]swarm.Address, error)
```

Comment:
this is a duplicate interface of what is already defined in `hive_test.go`.
I would be happy if this method can be extracted to a separate interface which can be then reused in the test too.

Reply:
It's not the same, but it is similar. I have switched to using this interface in tests as well. I am not sure if we need any additional granularity, since `Store` seem to implement the whole interface anywayz.

Reply:
I meant that the method signature is the same as used in the hive test file.
It would be nice, if Overlay and Multiaddresses are broken down into separate interfaces which are referenced by this interface. Because then why bother breaking down `GetPutter` either with the same convention?
Also, it might be benefical to have `addressbook.New` just return `addressbook.Interface` type instead of `GetPutter`, and get rid of the interface cast in the test case

Reply:
Ups, my bad, forgot to delete the cast thanks! The constructor is already returning `Interface`. I am not sure if we need this granularity of `GetPutter` etc.. I would say it makes sense if it reduces are job mocking, and maybe saves some components of unnecessary functionality, but in be I don't see much benefit at the moment, tbh. Since you introduced `Getter`, `Putter` and `GetPutter`, feel free to refactor them any way you like, I like both approaches, but personally I would even just leave the `Interface` for now and delete the more granular ones.